resources/js/components/utils/AccessBarrier.tsx

```diff
@@ -47,29 +47,17 @@ export const AccessBarrier: ParentComponent<AccessBarrierProps> = (props) => {
       Fallback: () => <p>Nie masz uprawnień do tego zasobu</p>,
       roles: [],
       Error: () => <Navigate href="/login" />,
-      Pending: () => (
-        <div class="h-screen flex justify-center items-center">
-          <MemoLoader size={300} />
-        </div>
-      ),
+      Pending: MemoLoader,
     },
     props,
   );
   const [queryBarrierProps, localProps] = splitProps(merged, ["Error", "Pending"]);
-
-  const queryClient = useQueryClient();
-
-  const facilityId = () =>
-    queryClient
-      .getQueryData<FacilityResource[]>(System.facilitiesQueryOptions().queryKey)
-      ?.find(({url}) => url === localProps.facilityUrl)?.id;
+  const facilitiesQuery = createQuery(System.facilitiesQueryOptions);
+  const facilityId = () => facilitiesQuery.data?.find(({url}) => url === localProps.facilityUrl)?.id;
```

Comment:
Są użycia bez facilityUrl, nie? (admin pages) Wtedy powinno pomijać niepotrzebne facilitiesQuery.

A druga rzecz, że może user getStatus API powinno brać facility url? Wygląda nieoptymalnie, że frontend musi ściągać wszystkie facilities żeby ustalać id.

Reply:
Ogólnie zapytanie do facilities jest inicjowane bardzo wcześnie, w InitializeTanstackQuery, a potem te query są skonfigurowane tak że cache'ują się raz na zawsze, więc to stworzenie facilitiesQuery nic nie kosztuje praktycznie, a na pewno nie robi zapytania (jest refetchOnMount:false czy coś takiego). Tych facilities to będzie tak rzędu kilka, więc to jest OK że zawsze je mamy ściągnięte, a przy tym to query to najprostszy sposób dostania się do nich. No i nadal jest możliwość inwalidacji gdy admin zmieni listę, albo jednak ustawienia jakiegoś timeoutu po którym zostaną przeładowane.

Reply:
Do przeżycia, ale nieidealne, że refresh /admin/users ściąga niepotrzebne dane. Ideowo bym wolał, żeby dane się ściągały jak najwcześniej, ale jednocześnie były mocno przywiązane do swoich użyć. Np. jak się usuwa kod który potrzebuje dane, to fajnie jak od razu fetch sam znika. Bo inaczej potem z czasem się akumulują jakieś niepotrzebne blokujące i nieblokujące fetche.

Reply:
A tutaj to chyba łatwo możesz obsłużyć ifem na facilityUrl.

Reply:
Akurat konkretnie facilities to jest coś co zakładamy że ściągamy zupełnie zawsze, chociażby dlatego że w nagłówku strony wyświetlamy nazwę facility albo dropdown do wyboru facilities. Więc nie ściągać facilities można by tylko dla użytkownika, który do żadnego facility nie jest przypisany, czyli bardzo rzadko albo nigdy (globalny admin też może mieć poza tym placówki).

Drugą taką rzeczą jest user status bo tam są uprawnienia do oglądania jakiejkolwiek strony. Więc podobnie, createQuery do tych uprawnień nic nie kosztuje bo one najpewniej i tak już są w cache. Tylko tam jest komplikacja że czasem podaje się też facility id w tym query a czasem nie.

Z kolei if nie jest aż taki łatwy bo query nie powinno się tworzyć dynamicznie, a facilityUrl jest teoretycznie reaktywne :) Myślę naprawdę że w tym miejscu można to odpuścić, szczególnie że InitializeTanstackQuery konkretnie pisze że te rzeczy są pobrane od razu.